timesketch/ui/graphs/graph-view.component.ts

```diff
@@ -0,0 +1,130 @@
+import {Component, Input} from '@angular/core'
+
+import {CytoscapeSettings} from './cytoscape-settings.component'
+
+export type GraphViewState = {
+  type: 'loading'
+} | {
+  type: 'empty'
+} | {
+  type: 'ready'
+  elements: Cy.ElementsDefinition | Cy.ElementDefinition[]
+}
+
+@Component({
+  selector: 'ts-graphs-graph-view',
+  templateUrl: './graph-view.component.html',
+})
+export class GraphViewComponent {
+  // tslint:disable-next-line:no-unused-variable
+  @Input() state: GraphViewState
+  settings: CytoscapeSettings = {
+    // interaction options:
+    minZoom: 1e-50,
+    maxZoom: 1e50,
+    zoomingEnabled: true,
+    userZoomingEnabled: true,
+    panningEnabled: true,
+    userPanningEnabled: true,
+    boxSelectionEnabled: true,
+    selectionType: 'single',
+    touchTapThreshold: 8,
+    desktopTapThreshold: 4,
+    autolock: false,
+    autoungrabify: false,
+    autounselectify: false,
+    // rendering options:
+    headless: false,
+    styleEnabled: true,
+    hideEdgesOnViewport: false,
+    hideLabelsOnViewport: false,
+    textureOnViewport: false,
+    motionBlur: false,
+    motionBlurOpacity: 0.2,
+    wheelSensitivity: 1,
+    pixelRatio: ('auto' as any),
+  }
+  layout = {
+    name: 'cose',
+    animate: true,
+    animationThreshold: 250,
+    animationDuration: 1000,
+    refresh: 20,
+    fit: true,
+    padding: 30,
+    boundingBox: undefined,
+    randomize: true,
+    componentSpacing: 200,
+    nodeRepulsion: (node) => 400000,
+    nodeOverlap: 10,
+    idealEdgeLength: (edge) => 10,
+    edgeElasticity: (edge) => 100,
+    nestingFactor: 5,
+    gravity: 50,
+    numIter: 1000,
+    initialTemp: 200,
+    coolingFactor: 0.95,
+    minTemp: 1.0,
+    useMultitasking: true,
+  }
+  style = [
```

Comment:
Can this be moved to CSS?

Reply:
No, this is Cytoscape-specific.

Reply:
I will add type annotation for that too.